gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonLoggingInterceptor.java

```diff
@@ -0,0 +1,116 @@
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.api.gax.httpjson;
+
+import static com.google.api.gax.logging.LoggingUtils.logRequest;
+import static com.google.api.gax.logging.LoggingUtils.logResponse;
+import static com.google.api.gax.logging.LoggingUtils.recordResponseHeaders;
+import static com.google.api.gax.logging.LoggingUtils.recordResponsePayload;
+import static com.google.api.gax.logging.LoggingUtils.recordServiceRpcAndRequestHeaders;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.httpjson.ForwardingHttpJsonClientCall.SimpleForwardingHttpJsonClientCall;
+import com.google.api.gax.httpjson.ForwardingHttpJsonClientCallListener.SimpleForwardingHttpJsonClientCallListener;
+import com.google.api.gax.logging.LogData;
+import com.google.api.gax.logging.LoggerProvider;
+import java.util.HashMap;
+import java.util.Map;
+
+@InternalApi
+public class HttpJsonLoggingInterceptor implements HttpJsonClientInterceptor {
+
+  private static final LoggerProvider LOGGER_PROVIDER =
+      LoggerProvider.forClazz(HttpJsonLoggingInterceptor.class);
+
+  @Override
+  public <ReqT, RespT> HttpJsonClientCall<ReqT, RespT> interceptCall(
+      ApiMethodDescriptor<ReqT, RespT> method,
+      HttpJsonCallOptions callOptions,
+      HttpJsonChannel next) {
+
+    String endpoint = ((ManagedHttpJsonChannel) next).getEndpoint();
+
+    return new SimpleForwardingHttpJsonClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+
+      LogData.Builder logDataBuilder = LogData.builder();
+
+      @Override
+      public void start(
+          HttpJsonClientCall.Listener<RespT> responseListener, HttpJsonMetadata headers) {
+        recordServiceRpcAndRequestHeaders(
+            null,
```

Comment:
Can you add a small comment why this param is intentionally set to null here?

Reply:
[091ab38](https://github.com/googleapis/sdk-platform-java/pull/3403/commits/091ab3890033231e2808c7cb5acf21a36b7ef66d)